app.js

```diff
@@ -9,11 +8,71 @@ class Dashboard extends React.Component {
     /**
      * @inheritdoc
      */
+    constructor(props) {
+        super(props);
+        this.psnrTestResultTransformFunc = jsonData => {
+            const extractData = (json, yAxisFieldName) =>
+                json.reduce((currData, currRow) => {
+                    const field = currRow[yAxisFieldName];
+
+                    currData.push([ currRow.buildNum, field ]);
+
+                    return currData;
+                }, []);
+
+            // The frame data needs to be transformed to the percentage
+            //  of all the frames, so we need another helper
+            const extractPctData
+                = (json,
+                        yAxisNumeratorFieldName,
+                        yAxisDenominatorFieldName) =>
+                    json.reduce((currData, currRow) => {
+                        const num = currRow[yAxisNumeratorFieldName];
+                        const denom = currRow[yAxisDenominatorFieldName];
+                        const val = num / denom;
+
+                        currData.push([ currRow.buildNum, val ]);
+
+                        return currData;
+                    }, []);
+
+            const psnrData = extractData(jsonData, 'psnr');
+            const frameSkipData
+                = extractPctData(jsonData, 'numSkippedFrames', 'totalFrames');
+            const frameFrozenData
+                = extractPctData(jsonData, 'numFrozenFrames', 'totalFrames');
+
+            return [
+                {
+                    name: 'psnr',
+                    data: psnrData
+                },
+                {
+                    name: 'skipped frame %',
+                    data: frameSkipData
+                },
+                {
+                    name: 'frozen frame %',
+                    data: frameFrozenData
+                }
+            ];
+        };
+    }
```

Comment:
This is pretty clever. At first I was kind of worried because maybe the relationship between this parser function and the chart is separated too much, but the more I read it the more I think it works well enough.